src/sourcegraph.d.ts

```diff
@@ -785,6 +785,37 @@ declare module 'sourcegraph' {
         ): Unsubscribable
     }
 
+    /**
+     * A query transformer alters a user's search query before executing a search.
+     *
+     * Query transformers allow extensions to define new search query operators and syntax, for example,
+     * by matching strings in a query (e.g. `go.imports:`) and replacing them with a regular expression or string.
+     */
+    export interface QueryTransformer {
+        /**
+         * Transforms a search query into another, valid query. If there are no transformations to be made
+         * the original query is returned.
+         *
+         * @param query A search query provided by a client.
+         */
+        transformQuery(query: string): string | Promise<string>
+    }
+
+    /**
+     * API for extensions to augment search functionality.
+     */
+    export namespace search {
+        /**
+         * Registers a query transformer.
+         *
+         * Multiple transformers can be registered. In that case, all transformers will be executed
```

Comment:
I think it is more idiomatic to say a transformation is applied, not executed. So replace "executed" with "applied".

Also, the order in which multiple transformers are applied is undefined (i.e., the extension API makes no guarantee). So I'd state that.